frontend/hub/collections/CollectionDetails.tsx

```diff
@@ -52,50 +52,78 @@ import { PageDetail } from '../../../framework/PageDetails/PageDetail';
-  const itemActions = useCollectionActions(() => void refresh());
+  const itemActions = useCollectionActions(() => void refresh(), true);
   const [collection, setCollection] = useState<CollectionVersionSearch | undefined>(undefined);
+  const [collections, setCollections] = useState<CollectionVersionSearch[]>([]);
+  const [collectionError, setCollectionError] = useState<JSX.Element | undefined>(undefined);
+  const request = useGetRequest<HubItemsResponse<CollectionVersionSearch>>();
 
-  let collectionError: JSX.Element | undefined = undefined;
+  const navigate = usePageNavigate();
 
   function setVersionParams(version: string) {
     setSearchParams((params) => {
       params.set('version', version);
       return params;
     });
   }
-  // load all collections versions belong to the repository
-  const collectionsResult = useGet<HubItemsResponse<CollectionVersionSearch>>(
-    hubAPI`/v3/plugin/ansible/search/collection-versions/?name=${
-      searchParams.get('name') || ''
-    }&namespace=${searchParams.get('namespace') || ''}&repository_name=${
-      searchParams.get('repository') || ''
-    }&order_by=-version`
-  );
 
-  if (
-    collectionsResult.error ||
-    (collectionsResult.isLoading == false &&
-      collectionsResult.data &&
-      collectionsResult.data.data.length == 0)
-  ) {
-    collectionError = (
-      <AwxError error={collectionsResult.error || { name: 'not found', message: t('Not Found') }} />
-    );
-  }
+  const name = searchParams.get('name') || '';
+  const namespace = searchParams.get('namespace') || '';
+  const repository = searchParams.get('repository') || '';
+  const redirectIfEmpty = searchParams.get('redirectIfEmpty') || '';
 
-  const collections = collectionsResult.data ? collectionsResult.data.data : [];
+  useEffect(() => {
+    if (!(name && namespace && repository)) {
+      setCollectionError(<AwxError error={{ name: 'not found', message: t('Not Found') }} />);
+    }
+  }, [name, namespace, repository, setCollectionError, t]);
+
+  useEffect(() => {
+    try {
+      void (async function () {
+        const res = await request(hubAPI`/v3/plugin/ansible/search/collection-versions/`, {
+          name,
+          namespace,
+          repository_name: repository,
+        });
+
+        if (res.data.length == 0) {
+          if (redirectIfEmpty) {
+            navigate(HubRoute.Collections);
+          } else {
+            setCollectionError(<AwxError error={{ name: 'not found', message: t('Not Found') }} />);
+          }
+        }
+
+        if (redirectIfEmpty) {
+          const newParams = new URLSearchParams(searchParams.toString());
+
+          // Set a new query parameter or update existing ones
+          newParams.set('redirectIfEmpty', '');
+
+          setSearchParams(newParams);
+        }
+
+        setCollections(res.data);
+      })();
+    } catch (error) {
+      setCollectionError(<AwxError error={{ name: 'not found', message: t('Not Found') }} />);
```

Comment:
this will never happen without `await`, the async function just returns a rejected promise on failure